### DISCUSSION TITLE
Veramo example including Holder, Issuer and Verifier (or Verifiable Credentials' Request and Response)

### DISCUSSION THREAD

**Discussion by amal-abid05** [decentralized-identity/veramo#533] [Q&A]
Hi,
I failed to find a demo code that includes Holder, Issuer and Verifier , or even Verifiable Credentials' Request and Response.
I am a beginner, and it is very difficult for me to use the API without examples.
With uPort, the documentation was very clear, now I feel lost.
https://developer.uport.me/credentials/login
I really need your help please.
Thank you.

**Comment by mirceanis** [comment]
Hi, I'm afraid there is no example describing all you are looking for (yet). Only fragments.

I know this is far from being an answer, but your best bet at the moment is to sniff through the test suites like [this one](https://github.com/uport-project/veramo/blob/988c76c46d391f3b76499ff141bdefe21e729c4a/__tests__/shared/handleSdrMessage.ts) and [this one](https://github.com/uport-project/veramo/blob/988c76c46d391f3b76499ff141bdefe21e729c4a/__tests__/shared/verifiableData.ts).

That being said, we appreciate contributions, so if you do end up writing this demo, we would be happy to showcase it.

**Reply by amal-abid05** [reply]
Thank you, I will try to use these examples.

**Comment by jasheal** [comment]
Hi @amal-abid05,

Thanks for checking out Veramo! Once thing to note is although Veramo is an evolution of uPort it is not a direct 1-1 replacement of the full uPort stack.  uPort was part application, part infrastructure and part protocol whereas Veramo is mostly protocols.

If there's something in particular you are trying to do the best approach would be to describe your use case here and we will try to help as much as we can. Also if you haven't already, make sure to checkout our docs here https://veramo.io/docs/basics/introduction
